MHVLib/Connection/MHVConnection.m

```diff
@@ -402,6 +445,20 @@ - (void)refreshSessionCredentialWithCompletion:(void(^_Nullable)(NSError *_Nulla
 
 - (void)refreshTokenAndReissueRequest:(MHVHttpServiceRequest *)request
 {
+    // If it already refreshed the token for this request, don't retry since refreshing again won't fix the problem
+    if (request.hasRefreshedToken)
+    {
+        MHVLOG(@"The refreshed session authorization is invalid or expired.");
+        
+        if (request.completion)
+        {
+            request.completion(nil, [NSError error:[NSError MHVUnauthorizedError]
+                                   withDescription:@"The refreshed session authorization is invalid or expired."]);
```

Comment:
We should add to the description that an attempt to refresh the token was made but failed.